### DISCUSSION TITLE
Converting PTracer into Volpath

### DISCUSSION THREAD

**Discussion by Ghiraa** [mitsuba-renderer/mitsuba3#1071] [Q&A]
Hello,

I'm currently conducting an experiment to observe the differences in light interaction with a sensor, specifically with and without water interference. My challenge is that my custom light source functions correctly only when using a ptracer integrator and fails to operate with a volpath, leaving me uncertain about how to incorporate participating media under the ptracer setup.

Is there a workaround or modification that would allow the ptracer integrator to account for participating media? Alternatively, could someone point me towards a section in the code where a similar implementation exists for reference? Or, is the volpath integrator the only option for modeling participating media accurately?

Any guidance or suggestions would be greatly appreciated. Thank you!

**Comment by njroussel** [comment]
Hi @Ghiraa 

I don't believe there is an easy hack to achieve this, you need to re-implement an integrator that can handle participating media with light tracing.
